### DISCUSSION TITLE
Error using CHECKPOINT option

### DISCUSSION THREAD

**Discussion by Silvia-Caligari** [idaholab/moose#20711] [Q&A Modules: General] [answered]
Hi,
I'm trying to do simulations on a university cluster using the checkpoint moose option and I obtain this error:
```
Make sure LIBMESH_HAVE_XDR is defined at build time

The XDR interface is not available in this installation

[11] /home/caligari/AVFLOW/moose/scripts/../libmesh/src/utils/xdr_cxx.C, line 147, compiled

PMIX ERROR: UNREACHABLE in file ../../../../../../../src/openmpi-4.0.5/opal/mca/pmix/pmix3x/pmix/src/server/pmix_server.c at line 2193
```

Is any solution available?

Thank you

**Comment by GiudGiud** [comment, marked as answer]
Hello

Please see this answer https://github.com/idaholab/moose/discussions/20479#discussioncomment-2311198
You need to install libtirpc-devel on your cluster.

If you cant do that and need to use checkpoint anyway please see this answer:
https://github.com/idaholab/moose/discussions/20479#discussioncomment-2311796

Guillaume

**Reply by Silvia-Caligari** [reply]
Thank you for your answer.
Can you please link me any information about the installation on cluster? 
Thank you very much

**Reply by GiudGiud** [reply]
This is typically installed by your cluster administrator. You should get in touch with your university cluster administrators

**Reply by Silvia-Caligari** [reply]
Thank you for your answer,
we have just installed the library, but how can we give to lib mesh the path of the library?

**Reply by GiudGiud** [reply]
Did you install it in a non-standard place?
If you did not, libmesh will just know

**Reply by Silvia-Caligari** [reply]
Yes, it is in a non-std place as our system does not provide that library. How and where can I set the paths to a non-std place?

**Reply by Silvia-Caligari** [reply]
We tried using:

export libmesh_optional_INCLUDES=/home/caligari/AVFLOW/libtirpc-1.3.2/bin/include/tirpc

export CPPFLAGS=-I/home/caligari/AVFLOW/libtirpc-1.3.2/bin/include/

but they didn't work. Do you have any suggestions?

**Reply by GiudGiud** [reply]
@milljm 
this is definitely what you need to do, but there may be subtleties

**Comment by Silvia-Caligari** [comment]
We tried using:

export libmesh_optional_INCLUDES=/home/caligari/AVFLOW/libtirpc-1.3.2/bin/include/tirpc

export CPPFLAGS=-I/home/caligari/AVFLOW/libtirpc-1.3.2/bin/include/


but they didn't work. Do anyone have any suggestions?

**Reply by GiudGiud** [reply]
Hello

Is this solved?
Or are you happy enough with the `binary = false` workaround?
Btw you dont want to discard existing CPP_FLAGS so you may want to have:
`export CPPFLAGS=$CPPFLAGS -I/home/caligari/AVFLOW/libtirpc-1.3.2/bin/include/`

Guillaume
